utils/interfaces/_core.py

```diff
@@ -196,71 +200,75 @@ class BaseValidation(object):
     _interface = None  # which interface will be validated
     is_success_on_expiry = False  # if validation is still pending at end of procees, is it a success?
     path_filters = None  # Can be a string, or a list of string. Will perfom validation only on path in it.
+    system_test_error = None  # if something bad happen, the excpetion will be stored here
 
     def __init__(self, message=None, request=None, path_filters=None):
-        self.expected_timeout = None
-        self.message = message
-        self._closed = threading.Event()
-        self._is_success = None
-
-        if path_filters is not None:
-            self.path_filters = path_filters
+        try:
+            self.expected_timeout = None
+            self.message = message
+            self._closed = threading.Event()
+            self._is_success = None
 
-        if isinstance(self.path_filters, str):
-            self.path_filters = [self.path_filters]
+            if path_filters is not None:
+                self.path_filters = path_filters
 
-        if self.path_filters is not None:
-            self.path_filters = [re.compile(path) for path in self.path_filters]
+            if isinstance(self.path_filters, str):
+                self.path_filters = [self.path_filters]
 
-        if request is not None:
-            user_agent = [v for k, v in request.request.headers.items() if k.lower() == "user-agent"][0]
-            self.rid = user_agent[-36:]
-        else:
-            self.rid = None
+            if self.path_filters is not None:
+                self.path_filters = [re.compile(path) for path in self.path_filters]
 
-        self.frame = None
-        self.calling_method = None
-        self.is_xfail = False
+            if request is not None:
+                user_agent = [v for k, v in request.request.headers.items() if k.lower() == "user-agent"][0]
+                self.rid = user_agent[-36:]
```

Comment:
what is the meaning of -36?

Reply:
* All requests sent to the weblog has a `user-agent` header with the form `<whatever> rid/<36 uppercase chars>`.
* `<36 uppercase chars>` is the request ID (aka `rid` in the code) that identify the requests.
* in python `array[-36:]` takes the last `N` elements of the array
* 
=> it extracts the rid